app/models/metric.rb

```diff
@@ -133,19 +133,21 @@ def unit=(value)
     self[:unit] = value unless child?
   end
 
+  def type
+    @type ||= method_metric? ? :method : :metric
+  end
+
   alias method_metric? child?
 
   def to_xml(options = {})
     xml = options[:builder] || ThreeScale::XML::Builder.new
 
-    metric_or_method = method_metric? ? 'method' : 'metric'
-
-    xml.__send__(:method_missing, metric_or_method) do |xml|
+    xml.__send__(:method_missing, type) do |xml|
       xml.id_           id unless new_record?
       xml.name          name # As of February 2014 this is deprecated and should be removed
       xml.system_name   name
       xml.friendly_name friendly_name
-      xml.service_id    (backend_api_metric? ? nil : owner_id)
+      xml.service_id    service_owner&.id
```

Comment:
Hm... In theory this could happen, I guess. I think, however, that this specific line will not be affected, because it was already using `owner_id`, and I guess if there is `owner_id`, there is `owner`.

But in general, I guess it's a good point, yes. The feature depends on metrics having an `owner`, but indeed, some old versions might not have it.

Where did you run this query on? SaaS?
Was some backfill script run on it manually? Otherwise, how is it possible? :) Maybe there is already some dark logic to protect from missing owners?

Reply:
I don't remember for sue but I think SaaS was good already when I checked that in the past. Maybe somebody run a query to fix these when `owner` was introduced.